AI Agent: add documentation for structured outputs/JSON mode

maff opened the pull request:
## Description

Documentation for structured output support in AI Agent connector implemented in https://github.com/camunda/connectors/issues/4723.

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [ ] This is on a **specific schedule** and the assignee will coordinate a release with the Documentation team. (create draft PR and/or add `hold` label)
- [x] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [x] My changes are for **an upcoming minor release** and are in the `/docs` directory (version 8.8).
- [ ] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [ ] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [ ] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

maff commented:
This will be released with the next alpha6 - not sure if we want to hold this until there or mark it as available from the next alpha somewhere in the text?

mesellings commented:
Added a docs issue for tracking: https://github.com/camunda/camunda-docs/issues/6027

mesellings reviewed (dismissed):
Great work as always @maff, approving 🚀 with some notes:
- I've made some rewording edits etc as normal
- I've removed the badges for now - this is a great idea, but will quickly become unsustainable and pollute the UX over time/releases across the docs - as alphas are not the 'finished version' of the docs, I don't think we need to highlight any doc changes in such a specific way. However, thinking of the user as you are here 👏 , we can highlight this change in the release notes (is this a change in the connector, or just the docs?).
- I've kept the react component in there for now, but we could/should remove it really if we aren't going to use it here?

maff commented:
Thanks a lot @mesellings! 

> * I've removed the badges for now - this is a great idea, but will quickly become unsustainable and pollute the UX over time/releases across the docs - as alphas are not the 'finished version' of the docs, I don't think we need to highlight any doc changes in such a specific way. However, thinking of the user as you are here 👏 , we can highlight this change in the release notes (is this a change in the connector, or just the docs?).
> * I've kept the react component in there for now, but we could/should remove it really if we aren't going to use it here?

I agree that these badges will clutter the docs, and as we're approaching alpha6 they are not relevant anymore, so perfectly fine to remove them 👍 I also removed the component now. Could you re-review?

It's a new feature on the connector so it should definitely be part of the release notes.

mesellings reviewed (approved):
Perfect - approved! 🚀
